server/src/test/java/org/elasticsearch/snapshots/mockstore/MockRepository.java

```diff
@@ -377,6 +362,9 @@ public void writeBlob(String blobName, InputStream inputStream, long blobSize, b
             public void writeBlobAtomic(final String blobName, final InputStream inputStream, final long blobSize,
                                         final boolean failIfAlreadyExists) throws IOException {
                 final Random random = RandomizedContext.current().getRandom();
+                if (blobName.startsWith("index-") && blockOnWriteIndexFile) {
```

Comment:
This is required to fix #25281. Sorry for mixing this in here, but I think the fact that we now how to allow overwrites in snapshot finalization and its effects on this are probably easier understood in one go (there was some discussion on this change in https://github.com/elastic/elasticsearch/pull/36927 and I think the concerns from there are addressed here now in the explanations added to the snapshot finalization).